rosbag2_cpp/src/rosbag2_cpp/info.cpp

```diff
@@ -127,20 +171,112 @@ std::vector<std::shared_ptr<rosbag2_service_info_t>> Info::read_service_info(
                 "Failed to deserialize message from " + bag_msg->topic_name + " !");
       }
 
-      switch (msg.event_type) {
-        case service_msgs::msg::ServiceEventInfo::REQUEST_SENT:
-        case service_msgs::msg::ServiceEventInfo::REQUEST_RECEIVED:
-          service_process_info[bag_msg->topic_name]->request[msg.client_gid].emplace(
-            msg.sequence_number);
-          break;
-        case service_msgs::msg::ServiceEventInfo::RESPONSE_SENT:
-        case service_msgs::msg::ServiceEventInfo::RESPONSE_RECEIVED:
-          service_process_info[bag_msg->topic_name]->response[msg.client_gid].emplace(
-            msg.sequence_number);
-          break;
+      if (action_topic_type != TopicsInAction::Unknown) {
+        auto action_name = action_topic_name_to_action_name(bag_msg->topic_name);
+
+        // Handle action service event topic
+        switch (msg.event_type) {
+          case service_msgs::msg::ServiceEventInfo::REQUEST_SENT:
+          case service_msgs::msg::ServiceEventInfo::REQUEST_RECEIVED:
+            {
+              if (action_topic_type == TopicsInAction::SendGoalEvent) {
+                action_process_info[action_name]->
+                send_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
+              } else if (action_topic_type == TopicsInAction::GetResultEvent) {
+                action_process_info[action_name]->
+                get_result_service.request[msg.client_gid].emplace(msg.sequence_number);
+              } else {
+                // TopicsInAction::CancelGoalEvent
+                action_process_info[action_name]->
+                cancel_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
+              }
```

Comment:
We can simplify by using a reference to the current action
```suggestion
               if (action_interface_type == TopicsInAction::SendGoalEvent) {
                action->send_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
              } else if (action_interface_type == TopicsInAction::GetResultEvent) {
                action->get_result_service.request[msg.client_gid].emplace(msg.sequence_number);
              } else {
                // TopicsInAction::CancelGoalEvent
                action->cancel_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
              }
```

Reply:
@Barry-Xu-2018 I meant that if rename `action_service_info` to the shorter name **action** we can make code inside if-else statements more readable and fit in one line in most cases.
for instance instead of the 
```cpp
              if (action_interface_type == ActionInterfaceType::SendGoalEvent) {
                action_service_info->
                send_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
              } else if (action_interface_type == ActionInterfaceType::GetResultEvent) {
                action_service_info->
                get_result_service.request[msg.client_gid].emplace(msg.sequence_number);
              } else {
                // TopicsInAction::CancelGoalEvent
                action_service_info->
                cancel_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
              }
```
it could be 
```cpp
              if (action_interface_type == ActionInterfaceType::SendGoalEvent) {
                action->send_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
              } else if (action_interface_type == ActionInterfaceType::GetResultEvent) {
                action->get_result_service.request[msg.client_gid].emplace(msg.sequence_number);
              } else {
                // TopicsInAction::CancelGoalEvent
                action->cancel_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
              }
```

Reply:
Got it. I will update it.